src/ai_player.rs

```diff
@@ -45,12 +47,34 @@ impl AiPlayer {
     fn play_single_turn_with_engines_order_memory<GridType: Grid>(&mut self, level_description: &LevelDescription, grid: &mut GridType) {
         for i in 0..self.engines.len() {
             self.current_engine = (self.current_engine + i) % self.engines.len();
-            if self.engines[self.current_engine].play_one_turn(level_description, grid) {
+            if process_lines(level_description, grid, &*self.engines[self.current_engine]) {
                 return;
             }
         }
     }
     pub fn pause_play(&mut self) {
         self.is_active = false
     }
+}
+
+fn process_lines<GridType: Grid>(level_description: &LevelDescription, grid: &mut GridType, strategy: &dyn SolvingStrategy) -> bool {
```

Comment:
Please either use `///` to document the meaning of the returned boolean, or use a custom enum instead